e3/net/token.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import, division, print_function
+
+import base64
+import calendar
+import json
+from datetime import datetime
+
+
+def read_jwt(filename):
+    """Read a jwt token.
+
+    :param filename: jwt token path
+    :type filename: str
+
+    :return: a tuple containing the jwt raw content and the decoded payload
+    :rtype: (str, dict)
+    """
+    with open(filename, 'rb') as f:
+        jwt = f.read()
+
+    # Extract the payload
+    signing_part, _ = jwt.rsplit(b'.', 1)
+    _, payload = signing_part.split(b'.', 1)
+
+    # Add required padding
+    rem = len(payload) % 4
+    if rem > 0:
+        payload += b'=' * (4 - rem)
+    return jwt, json.loads(base64.b64decode(payload).decode('utf-8'))
+
+
+def utc_timestamp():
+    """Return the number of seconds since epoch UTC."""
+    d = datetime.utcnow()
+    return calendar.timegm(d.utctimetuple())
+
+
+def is_valid(token):
+    """Return true if the token is still valid, false otherwise."""
```

Comment:
use proper docstring

Reply:
technically that's token_payload for the parameter not the full token. what about accepting here a full token and doing the decoding in the is_valid function ? The token might not always come from a file